examples/linear_model/plot_ard.py

```diff
@@ -1,119 +1,211 @@
-for one dimensional regression using polynomial feature expansion.
-Note the uncertainty starts going up on the right side of the plot.
-This is because these test samples are outside of the range of the training
-samples.
+# Author: Arturo Amor <david-arturo.amor-quiroz@inria.fr>
 
-"""
+# %%
+# Models robustness to recover the ground truth weights
+# =====================================================
+#
+# Generate synthetic dataset
+# --------------------------
+#
+# We generate a dataset where `X` and `y` are linearly linked: 10 of the
+# features of `X` will be used to generate `y`. The other features are not
+# useful at predicting `y`. In addition, we generate a dataset where `n_samples
+# == n_features`. Such a setting is challenging for an OLS model and leads
+# potentially to arbitrary large weights. Having a prior on the weights and a
+# penalty alleviates the problem. Finally, gaussian noise is added.
+
+from sklearn.datasets import make_regression
+
+X, y, true_weights = make_regression(
+    n_samples=100,
+    n_features=100,
+    n_informative=10,
+    noise=8,
+    coef=True,
+    random_state=42,
+)
 
-import numpy as np
+# %%
+# Fit the regressors
+# ------------------
+#
+# We now fit both Bayesian models and the OLS to later compare the models'
+# coefficients.
+
+import pandas as pd
+from sklearn.linear_model import ARDRegression, LinearRegression, BayesianRidge
+
+olr = LinearRegression().fit(X, y)
+brr = BayesianRidge(compute_score=True, n_iter=30).fit(X, y)
+ard = ARDRegression(compute_score=True, n_iter=30).fit(X, y)
+df = pd.DataFrame(
+    {
+        "Weights of true generative process": true_weights,
+        "ARDRegression": ard.coef_,
+        "BayesianRidge": brr.coef_,
+        "LinearRegression": olr.coef_,
+    }
+)
+
+# %%
+# Plot the true and estimated coefficients
+# ----------------------------------------
+#
+# Now we compare the coefficients of each model with the weights of
+# the true generative model.
 import matplotlib.pyplot as plt
-from scipy import stats
+import seaborn as sns
+from matplotlib.colors import SymLogNorm
+
+plt.figure(figsize=(10, 6))
+ax = sns.heatmap(
+    df.T,
+    norm=SymLogNorm(linthresh=10e-4, vmin=-80, vmax=80),
+    cbar_kws={"label": "coefficients' values"},
+    cmap="seismic_r",
+)
+plt.ylabel("linear model")
+plt.xlabel("coefficients")
+plt.tight_layout(rect=(0, 0, 1, 0.95))
+_ = plt.title("Models' coefficients")
 
-from sklearn.linear_model import ARDRegression, LinearRegression
+# %%
+# Due to the added noise, none of the models recover the true weights. Indeed,
+# all models always have more than 10 non-zero coefficients. Compared to the OLS
+# estimator, the coefficients using a Bayesian Ridge regression are slightly
+# shifted toward zero, which stabilises them. The ARD regression provides a
+# sparser solution: some of the non-informative coefficients are set exactly to
+# zero, while shifting others closer to zero. Some non-informative coefficients
```

Comment:
I am not sure about `non-informative` here. If coefficients are non-informative then the ground truth is zeros. Later, we speak about "order of magnitude" which is a bit confusing with zero-coefficient.

Reply:
I meant that some non-informative coefficients are still present and have large values, as large as the informative coefficients of the ground truth generating process. I am rephrasing this part for better clarity, thanks for pointing out this ambiguity.